### PR TITLE
Bump dependencies to remove notices with Ruby 2.4

### PR DESCRIPTION
**Pull request by renchap**
Ruby 2.4 unified Bignum and Fixnum into Integer.
This causes deprecation notices when running the tests. For example: https://travis-ci.org/grosser/fast_gettext/jobs/264417290

This commit bumps dependencies to versions than correctly handles Ruby 2.4:
- Rails from 5.0 to 5.1
- diff-lcs from 1.2 to 1.3

### CODE REVIEW

**Comment by renchap** [comment]
Note: the other deprecation notice is fixed in #95 but I wanted to make 2 PR, as I am a bit less confident about this one.

File: gemfiles/rails50.gemfile [hunk 1]
```diff
@@ -2,4 +2,4 @@ source "https://rubygems.org"
 
 gemspec path: "../"
 
-gem "rails", "~> 5.0.0"
+gem "rails", "~> 5.1.0"
```
Comment (grosser) [inline gemfiles/rails50.gemfile]: this is wrong cannot have rails 5.1 in a rails 5.0 gemfile

**Comment by grosser** [comment]
good idea to split up the PRs :)
... best do a new gemfile for rails 5.1 and then exclude 5.0 from ruby 2.4 like https://github.com/grosser/fast_gettext/blob/master/.travis.yml#L15-L17

**Comment by renchap** [comment]
Here you go. I also updated the dependancies on the other versions to the latest available.

**Comment by grosser** [comment]
thx!

**Pull request merged**